### DISCUSSION TITLE
Failed to get trophies for game I played

### DISCUSSION THREAD

**Discussion by marco120** [isFakeAccount/psnawp#36] [Q&A] [answered]
Hello, first of all thank you for this very nice work!
Though, even with such a nice doc and example, I'm really afraid to struggling so much to get the trophy list of my games from psnawp package.
Could you please help me a bit?
eg, for the PS5 game "Marvel's Spider-Man: Miles Morales" for which I got a platinium two weeks go, the trophy_group_summary indicate hidden_flag equal to none and earned_trophies is 0.
I admit I'm a bit confused by game_title and np_communication_id therefore, incode below I looped on different combination...

Here is the code:
```py
from psnawp_api import PSNAWP

# got from https://ca.account.sony.com/api/v1/ssocookie
psnawp = PSNAWP('<npsso-code>')

client = psnawp.me()
print(client.online_id)
print(client.account_id)

earned_trophies = client.trophy_summary().earned_trophies
print(earned_trophies)
print()

for trophy_title in client.trophy_titles(limit=1):
    print("Last trophy earned:")
    last_trophy_game_name = trophy_title.title_name
    print(f"last_trophy_game_name = {last_trophy_game_name}")
    print(trophy_title.last_updated_date_time)
print()

list_of_game = ['PPSA01411_00', 'PPSA01417_00', 'PPSA01418_00', 'PPSA01419_00', 'PPSA01460_00', 'PPSA01461_00' ]
for game in list_of_game:
    game_title = psnawp.game_title(game, 'me')
    trophy_groups_summary = game_title.trophy_groups_summary(platform = 'PS5')
    print(trophy_groups_summary.trophy_title_name)
    print(trophy_groups_summary.hidden_flag)
    trophy_set = trophy_groups_summary.earned_trophies
    print(trophy_set)
    game_title = psnawp.game_title(game, client.account_id)
    trophy_groups_summary = game_title.trophy_groups_summary(platform = 'PS5')
    print(trophy_groups_summary.hidden_flag)
    trophy_set = trophy_groups_summary.earned_trophies
    print(trophy_set)
```

Here is the result:

```
marc-spongebob
8451804021957075618
TrophySet(bronze=1667, silver=361, gold=85, platinum=11)

Last trophy earned:
last_trophy_game_name = Fall Guys
2023-02-23 13:51:47+00:00

Marvel's Spider-Man: Miles Morales
None
TrophySet(bronze=0, silver=0, gold=0, platinum=0)
None
TrophySet(bronze=0, silver=0, gold=0, platinum=0)
Marvel's Spider-Man: Miles Morales
None
TrophySet(bronze=0, silver=0, gold=0, platinum=0)
None
TrophySet(bronze=0, silver=0, gold=0, platinum=0)
Marvel's Spider-Man: Miles Morales
None
TrophySet(bronze=0, silver=0, gold=0, platinum=0)
None
TrophySet(bronze=0, silver=0, gold=0, platinum=0)
Marvel's Spider-Man: Miles Morales
None
TrophySet(bronze=0, silver=0, gold=0, platinum=0)
None
TrophySet(bronze=0, silver=0, gold=0, platinum=0)
Marvel's Spider-Man: Miles Morales
None
TrophySet(bronze=0, silver=0, gold=0, platinum=0)
None
TrophySet(bronze=0, silver=0, gold=0, platinum=0)
Marvel's Spider-Man: Miles Morales
None
TrophySet(bronze=0, silver=0, gold=0, platinum=0)
None
TrophySet(bronze=0, silver=0, gold=0, platinum=0)
```

**Comment by isFakeAccount** [comment, marked as answer]
Try using this method here.

https://psnawp.readthedocs.io/en/latest/psnawp_api.models.html#psnawp_api.models.client.Client.trophy_groups_summary

You can use this website to get the title id. If you live in America, then you only need the UP region.

If you want to search for the trophies you have earned, use the client class methods

If you want to check the trophies others have earned, use the user class methods.

The game title class is there for checking what trophies you can earn for a video game. This won't contain the stats related to the earned trophies.

**Reply by marco120** [reply]
thank you I managed to retrieve the list. I drop the code in case it can help someone:


game_id_search = psnawp.search()
title_id = game_id_search.get_title_id("Marvel's Spider-Man: Miles Morales")[-1]
print("title_id: " + title_id)
game_title = psnawp.game_title(title_id, client.account_id)
np_communication_id = game_title.np_communication_id
print("np_communication_id: " + np_communication_id)
trophy_groups_summary = client.trophy_groups_summary(np_communication_id, 'PS5', False)
earned_trophies = trophy_groups_summary.earned_trophies
print(earned_trophies)


>>>>>>>>>>> result:
title_id: PPSA01417_00
np_communication_id: NPWR20684_00
TrophySet(bronze=37, silver=10, gold=2, platinum=1)

>>>>>>>>>>>>>>>>>>>>

In doc, unless my mistake, it is not written that np_communication_id is a property of GameTitle class. 

Anyway, thank you for the help and the API! You made a happy one :)
